### DISCUSSION TITLE
I get a weird value back from a NRF24L01+ register

### DISCUSSION THREAD

**Discussion by elschmando** [micropython/micropython#11108] [Hardware & Peripherals]
Hello,
I try my luck with the NRF24L01+ transceiver and I'm getting a weird value back from register 0x17.
I just try to read if RX_EMPTY has data to fetch.

I do it throught this function:
`def read_reg(add):
    cs.on()
    send_add = bytearray()
    if (0x00 <= add <= 0x17) or (0x1c <= add <= 0x1d):
        send_add.append(read_conf_bin | add)
    else:
        send_add.append(add)
    
    cs.off()
    spi.write(send_add)
    data = spi.read(1)
    cs.on()
    
    return data`

and this command:
`test2 = read_reg(0x17)`
What does the @ mean?

**Comment by peterhinch** [comment]
It would help if you showed where the '@' occurred. My guess is that it's ASCII character '@' which is 0x40.

NRF24L01 is quite hard to use. I strongly recommend you start with [the official driver and test](https://github.com/micropython/micropython-lib/tree/master/micropython/drivers/radio/nrf24l01) before venturing into coding it directly.
